### PR TITLE
Pre-check dataset_id in register_and_submit_dataset

### PR DESCRIPTION
**Pull request by daniel-thom**
The code was not checking for proper status of the dataset in the project before registering the dataset. This was causing a lot of wasted time.

### CODE REVIEW

File: dsgrid/registry/project_registry_manager.py [hunk 1]
```diff
@@ -764,12 +772,8 @@ def _submit_dataset_and_register_mappings(
     ):
         logger.info("Submit dataset=%s to project=%s.", dataset_id, project_config.config_id)
         self._check_if_not_registered(project_config.config_id)
+        self._raise_if_not_unregistered(project_config, dataset_id)
```
Comment (elainethale) [inline dsgrid/registry/project_registry_manager.py:775]: Is this redundant with the line above?

**Review by elainethale — APPROVED** [review]
One minor question

File: dsgrid/registry/project_registry_manager.py (hunk 1 above)
Comment (daniel-thom) [reply dsgrid/registry/project_registry_manager.py:775]: Line 774 checks if the project_id is registered. Line 775 checks if the dataset is present in the project config and has an unregistered status.

Comment (elainethale) [reply dsgrid/registry/project_registry_manager.py:775]: I see that now--apologies for missing it the first time!

**Pull request merged**